### DISCUSSION TITLE
ERC721SeaDropContractOfferer  safeTransferFrom selector

### DISCUSSION THREAD

**Discussion by dcapitator** [ProjectOpenSea/seadrop#99] [General]
```
if (selector == IERC1155.safeTransferFrom.selector) {
            // Get the parameters.
            (
                address from,
                address to,
                uint256 id,
                uint256 amount,
                bytes memory transferData
            ) = abi.decode(data, (address, address, uint256, uint256, bytes));

            // Call safeTransferFrom.
            _safeTransferFrom(from, to, id, amount, transferData);
        }
```  
1-Is it used just because IERC721A has 2 safeTransferFrom signatures??
2-just seaport is allowed to operate secondary sales even after minting ??

```
    const safeTransferFrom1155Selector = "0xf242432a";

    const encodedParams = defaultAbiCoder.encode(
      ["address", "address", "uint256", "uint256", "bytes"],
      [token.address, AddressZero, 0, 1, []]
    );
    const data = safeTransferFrom1155Selector + encodedParams.slice(2);
 ```
 
 - in which case would seaport call ERC721SeaDrop with ERC1155.safeTransferFrom.selector ? 
 -why the checks of "from: token.address" ? and why it is not checked in fallback function erc1155contractOfferer(consistency)?
 
 - u missed the check for safeBatchTransferFrom ?
 -u override safeTransferFrom inside erc1155contractOfferer and erc1155PreapprovedConduit !! (either this or that no? )
 
 
@ryanio  thank  for clarifications

**Comment by ryanio** [comment]
the 1155 transfer in the 721 seadrop is a way to pass the mint quantity to the contract via seaport by using the "amount" field, so it is only used for that purpose

**Comment by dcapitator** [comment]
yes i figured it out , this is an old discussion . Thanks for the reply
